pkg/router/router_test.go

```diff
@@ -86,13 +89,17 @@ func TestMain(m *testing.M) {
 
 	var plugin router.Plugin
 
-	workdir, err := ioutil.TempDir("", "router")
+	var err error
+	workdir, err = ioutil.TempDir("", "router")
```

Comment:
this isn't necessary, you could keep the original

Reply:
I wanted `workdir` variable to be accessible by all the tests, that's why I moved it to the global scope.
The problem here is with the shadowing of the global variables when `:=` is used.
That is, if I keep it the way it was - the value of the global `workdir` won't change, Go will just create a new local variable and will set it (not the global).

Reply:
Put `workdir` to the global harness. This helped to revert this code back to how it looked before.